### ISSUE TITLE
Handling Tango attributes of invalid quality

### ISSUE THREAD

**Issue by LJBD** [taurus-org/taurus#387]
Hello!

I've stumbled across a possible bug (or unhandled exception): when creating a TaurusLabel and setting its model to a certain attribute that has invalid quality, I've received the following exception:
```python
push_event generated the following python exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/taurus/core/tango/tangoattribute.py", line 712, in push_event
    event.attr_value), None
  File "/usr/lib/python2.7/site-packages/taurus/core/tango/tangoattribute.py", line 409, in decode
    value = TangoAttrValue(pytango_dev_attr=attr_value, attr=self)
  File "/usr/lib/python2.7/site-packages/taurus/core/tango/tangoattribute.py", line 103, in __init__
    p.value = numpy.empty(shape, dtype=dtype)
UnboundLocalError: local variable 'shape' referenced before assignment
```

In order to investigate, I've created a simple Device with only one attribute and a command which changes the attribute's validity. Unfortunately, I haven't been able to see that exception again.
What I have discovered, though, is a possible regression: when an attribute's quality changes while there's a TaurusLabel displaying it, Taurus 3 starts to show a grey area. Taurus 4 continues to show the last value (see attached screenshot).
![taurus_validity_test](https://cloud.githubusercontent.com/assets/9516795/22330908/d747213a-e3c8-11e6-9453-d72db73db19f.jpg)
I guess that Taurus 3 behaviour is better since it lets a user know that something is wrong with an attribute.

So, to sum things up, there are 2 issues mentioned:
1. A weird error appearing for only one attribute with invalid quality.
2. A possible regression in displaying invalid attributes' values.

Best regards,
Łukasz Dudek

**Comment by cmft** [comment]
Hola @LJBD ,
I can reproduce your first error, it seems to be related with PyTango API change and a not protection in taurus.  I fix it in PR #388.

I can not reproduce the second reported error  "A possible regression in displaying invalid attributes' values."

In the same PR, I did some improvement in the TangoSchemeTest device (it is in taurus.core.tango.test.res module) to allow modify the attribute quality of an attribute (short_scalar_ro)

Could you try the next:

```
python -m taurus.qt.qtgui.display.tauruslabel YOUR/TangoSchemeTest/DEVICE/short_scalar_ro
```
and in jive use the command to set a different attribute quality, e.g.:
- invalid
- valid
-invalid


Best,
Carlos

**Comment by cpascual** [comment]
@LJBD : can you please confirm if the fix implemented in #388 fixes your problem?

**Comment by LJBD** [comment]
Yes, it solves both issues. Thank you very much!
